extensions/wikia/Discussions/utils/DiscussionsActivator.class.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+use Swagger\Client\ApiException;
+use Swagger\Client\Discussion\Api\SitesApi;
+use Swagger\Client\Discussion\Models\SiteInput;
+use Wikia\DependencyInjection\Injector;
+use Wikia\Service\Swagger\ApiProvider;
+
+class DiscussionsActivator {
+
+    const SERVICE_NAME = 'discussion';
+    const TIMEOUT = 5;
+    const SITE_NAME_MAX_LENGTH = 256;
+
+    private $cityId;
+    private $cityName;
+    private $cityLang;
+
+    private $sitesApi;
+    private $logger;
+
+    public function __construct( $cityId = null, $cityName = null, $cityLang = null ) {
+        $this->cityId = $cityId ?? F::app()->wg->CityId;
+        $this->cityName = $cityName ?? F::app()->wg->Sitename;
+        $this->cityLang = $cityLang ?? F::app()->wg->ContLang->getCode();
+
+        $this->sitesApi = $this->getSitesApi();
+        $this->logger = Wikia\Logger\WikiaLogger::instance();
+    }
+
+    public function activateDiscussions() {
+        if ( $this->isDiscussionsActive() ) {
+            return;
+        }
+
+        $siteInput = $this->getSiteInput();
+        try {
+            $this->sitesApi->createSite( $siteInput, F::app()->wg->TheSchwartzSecretToken );
+        } catch ( ApiException $e ) {
+            $this->logAndThrowError( $e );
+        }
+    }
+
+    private function getSiteInput() : SiteInput {
+        return new SiteInput(
+            [
+                'id' => $this->cityId,
+                'name' => substr( $this->cityName, 0, self::SITE_NAME_MAX_LENGTH ),
+                'language_code' => $this->cityLang
+            ]
+        );
+    }
+
+    public function isDiscussionsActive() {
+        try {
+            $this->sitesApi->getSite( $this->cityId );
+            return true;
+        } catch ( ApiException $e ) {
+            $this->logger->info( 'DISCUSSIONS Getting site caused an error',
+                [
+                    'siteId' => $this->cityId,
+                    'error' => $e->getMessage(),
+                ] );
+        }
+        return false;
+    }
+
+    private function getSitesApi() {
+        /** @var ApiProvider $apiProvider */
+        $apiProvider = Injector::getInjector()->get( ApiProvider::class );
+        /** @var SitesApi $api */
+        $api = $apiProvider->getApi( self::SERVICE_NAME, SitesApi::class );
+        $api->getApiClient()->getConfig()->setCurlTimeout( self::TIMEOUT );
+
+        return $api;
+    }
+
+    private function logAndThrowError( Exception $e ) {
+        $this->logger->error(
```

Comment:
I am actually using it in more than once place. See [here](https://github.com/Wikia/app/pull/11510/files#diff-1e926e0da49500c3ccffdb6634724100R59)